src/Gibe.LinkPicker.Umbraco/PropertyConverters/LinkPickerValueConverter.cs

```diff
@@ -40,9 +40,17 @@ public override object ConvertSourceToObject(PublishedPropertyType propertyType,
 
             var sourceString = source.ToString();
 
+            var umbracoHelper = new UmbracoHelper(UmbracoContext.Current);
+
             try
             {
-                var linkPicker = JsonConvert.DeserializeObject<Models.LinkPicker>(sourceString);
+                Models.LinkPicker linkPicker = JsonConvert.DeserializeObject<Models.LinkPicker>(sourceString);
+
+                if(linkPicker.Id > 0)
+                {
+                    linkPicker.Url = umbracoHelper.TypedContent(linkPicker.Id).Url;
```

Comment:
One potential downside is this will throw a NullReferenceException if the linked document is unpublished or deleted.

Maybe a null check in here and don't reset the URL in that case

Reply:
Good spot